src/pyenphase/envoy.py

```diff
@@ -168,18 +174,42 @@ def serial_number(self) -> str | None:
 
     async def probe(self) -> None:
         """Probe for model and supported features."""
-        for endpoint in (
-            URL_PRODUCTION_V1,
-            URL_PRODUCTION_JSON,
-            URL_PRODUCTION,
-        ):
-            try:
-                await self.request(endpoint)
-            except (json.JSONDecodeError, httpx.HTTPError) as e:
-                _LOGGER.debug("Production endpoint not found at %s: %s", endpoint, e)
-                continue
-            self._production_endpoint = endpoint
-            break
+        try:
+            production_json: dict[str, Any] = await self.request(URL_PRODUCTION)
+        except (json.JSONDecodeError, httpx.HTTPError) as e:
+            _LOGGER.debug("Production endpoint not found at %s: %s", URL_PRODUCTION, e)
+        else:
+            production: list[dict[str, str | float | int]] | None = production_json.get(
+                "production"
+            )
+            if production:
+                for type_ in production:
```

Comment:
What's the reason to pick `type_` over another variable name?

Reply:
type is reserved name in python